src/core/validations/username.ts

```diff
@@ -71,8 +71,9 @@ export function validateUsername(username: string): {
 }
 
 export function sanitizeUsername(str: string): string {
-  const sanitized = str
-    .replace(/[^a-zA-Z0-9_\[\] 🐈🍀]/gu, "")
-    .slice(0, MAX_USERNAME_LENGTH);
-  return sanitized.padEnd(MIN_USERNAME_LENGTH, "x");
+  let sanitized = str.replace(/[^a-zA-Z0-9_\[\] 🐈🍀]/gu, "");
+  if (sanitized.length < MIN_USERNAME_LENGTH) {
+    sanitized += Math.random().toString(36).substring(2, 2 + MIN_USERNAME_LENGTH - sanitized.length);
```

Comment:
Kind of a nitpick, but I think it would be easier to see at a glance what this function is doing if we generated exactly how many random characters we need... rather than creating a long string and truncating.